src/main/java/org/openstreetmap/atlas/checks/validation/intersections/SelfIntersectingPolylineCheck.java

```diff
@@ -144,4 +170,36 @@ else if (object instanceof Area)
     {
         return FALLBACK_INSTRUCTIONS;
     }
+
+    /**
+     * Returns true if adjacent {@link Edge} has identical lat,lng sequences
+     *
+     * @param polyline
+     *            the {@link PolyLine} being examined
+     * @return {@code true} if the any set of adjacent edges have identical geometries
+     */
+    private boolean hasDuplicateEdges(final PolyLine polyline)
+    {
+        final List<Segment> segments = polyline.segments();
+        final List<Segment> duplicates = new ArrayList<>();
+
+        // Loop through Polyline Segments
+        for (int i = 0; i < segments.size(); i++)
+        {
+            final Segment segment = segments.get(i);
+            final int adjacentEdge = i + 2;
+
+            // Check if segment exists elsewhere in List
+            if (segments.indexOf(segment) != segments.lastIndexOf(segment))
+            {
+                // If adjacent segment is the same value as our current, duplicate exists
+                if (adjacentEdge < segments.size() && segment.equals(segments.get(adjacentEdge)))
+                {
+                    duplicates.add(segment);
+                }
+            }
+        }
+
+        return duplicates.size() > 0;
```

Comment:
Test against empty should be sufficient here

Reply:
Wait you don't need to keep track of `duplicates`. Whenever a duplicate is found inside `for` loop, just return true. Otherwise, false could be returned after `for` loop.